### PR TITLE
Force comaptibility of artefacts across implementations

### PR DESCRIPTION
**Pull request by ale-linux**
This PR adds an integration test that asserts that signatures generated with the rust bbs crate can be verified by this package, and vice versa.

### CODE REVIEW

**Comment by ale-linux** [comment]
@swcurran I think this is a useful check to add. I have plans to extend the code in this repo to support the integration of BBS into more complex protocols, so this integration test should give us assurance that the implementation is still unchanged.

I suppose we might want to have the same tests in aries-bbssignature-rs?

File: integration/rust_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+	messagesBytes := [][]byte{
+		[]byte("message 1"),
+		[]byte("message 2"),
+		[]byte("message 3"),
+		[]byte("message 4"),
+		[]byte("message 5"),
+	}
+
+	bls := bbs.New()
```
Comment (yacovm) [inline integration/rust_test.go:67]: bls is bbs?

File: integration/rust_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	sm := re.FindStringSubmatch(string(output))
+	require.Len(t, sm, 3)
+
+	pkBytes, err := hex.DecodeString(string(sm[1]))
+	require.NoError(t, err)
+
+	sigBytes, err := hex.DecodeString(string(sm[2]))
+	require.NoError(t, err)
+
+	messagesBytes := [][]byte{
```
Comment (yacovm) [inline integration/rust_test.go:36]: maybe export to a constant / global variable?

File: integration/rust_signer/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        SignatureMessage::hash(b"message 1"),
+        SignatureMessage::hash(b"message 2"),
+        SignatureMessage::hash(b"message 3"),
+        SignatureMessage::hash(b"message 4"),
+        SignatureMessage::hash(b"message 5"),
+    ];
+
+    let args: Vec<String> = env::args().collect();
+
+    if args.len() == 1 {
```
Comment (yacovm) [inline integration/rust_signer/src/main.rs:26]: so, this branch is signing, the other branch is verifying?

**Comment by yacovm** [comment]
looks overall OK but one comment - maybe the messages should be randomized each time ? 
For example printed to a file, read from the file in the tests?

**Comment by yacovm** [comment]
or at least make the messages signed be "random" (hardcode a random string)

File: integration/rust_test.go (hunk 2 above)
Comment (ale-linux) [reply integration/rust_test.go:36]: Good point, done, and added comment that it needs to be equal to its rust counterpart

File: integration/rust_test.go (hunk 1 above)
Comment (ale-linux) [reply integration/rust_test.go:67]: done

File: integration/rust_signer/src/main.rs (hunk 3 above)
Comment (ale-linux) [reply integration/rust_signer/src/main.rs:26]: correct

**Comment by ale-linux** [comment]
Thx @yacovm. I fixed the requested changes inline. I'm not sure what randomising messages would add: we never do it anywhere else, and the way the scheme is designed, the message strings are hashed to random Zr.

**Comment by yacovm** [comment]
> Thx @yacovm. I fixed the requested changes inline. I'm not sure what randomising messages would add: we never do it anywhere else, and the way the scheme is designed, the message strings are hashed to random Zr.

Yes i realized it right as i stepped out for a few minutes and it was too late, you already responded.

**Review by yacovm — APPROVED** [review]

**Pull request merged**